test/api/application/delete_test.go

```diff
@@ -0,0 +1,32 @@
+package application
+
+import (
+	"testing"
+
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/test/api/client"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestApplicationDelete(t *testing.T) {
+	samples := Samples()
+	for _, r := range samples {
+		t.Run(r.Name, func(t *testing.T) {
+			// Create.
+			assert.Must(t, Create(&r))
+			rPath := client.Path(api.ApplicationRoot, client.Params{api.ID: r.ID})
+
+			// Try delete.
+			err := Client.Delete(rPath)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Check the it was deleted.
+			err = Client.Get(rPath, &r)
+			if err == nil {
+				t.Errorf("Exits, but should be deleted: %v", r)
+			}
```

Comment:
The responsibility of fixtures.go methods is to perform given action (and relay the error if returned), it doesn't know if the action should or should not raise the error.
The tests know what should be the result of calling fixtures methods (or direct API Get/Post/.. methods) and the test might not want rely on method's returned error, so it might do another cal to check the real result.
This allows both positive and negative tests and I believe, this _should not_ be part of fixtures.

Reply:
This comment makes me think we are in disagreement on the purpose of these fixtures. If you would like to update the `Create()` fixture to accept an `err` that represents the error that would be returned, that would be great. But the idea of the fixtures is to strengthen the verification for these core, always repeating actions, ie. everytime we create an application we make sure 1) we can GET it 2) we can LIST it 3) it's bucket is created as expected. This way, when some addon writes a test using these fixtures these verifications are happening for free.

It's important for fixtures to verify the thing that was supposed to happen happened.

Reply:
Probably the disagreement comes from different understanding what should be tested with different kinds of tests. The GET/LIST/BUCKET after Create makes sense to me for the API endpoint test, but e.g. the addon analyze test might not focus on the application creation action if it is bulletproof OK or not, the addon test should create the application (ideally by fixtures method) and if the Create method has not failed, use the application in its test.

Reply:
I'm not against adding a helper which will check the created/deleted/.. resource like application, I just don't see it as a mandatory part of the fixtures.go methods.